actionview/lib/action_view/template.rb

```diff
@@ -244,9 +244,17 @@ def supports_streaming?
     # This method is instrumented as "!render_template.action_view". Notice that
     # we use a bang in this instrumentation because you don't want to
     # consume this in production. This is only slow if it's being listened to.
-    def render(view, locals, buffer = nil, add_to_stack: true, &block)
+    def render(view, locals, buffer = nil, injected_locals: [], add_to_stack: true, &block)
       instrument_render_template do
         compile!(view)
+
+        if strict_locals? && injected_locals.present?
+          # It seems much easier to extract strict locals keys this way than parsing @strict_locals variable
+          strict_locals = view.method(method_name).parameters.select { |type, name| type.in?(%i[keyreq key]) }.map(&:last)
```

Comment:
`Module#method` is fairly costly, it would be best to find a cheaper way to check this. If necessary we can keep a list of the arguments on the view.